code/go/0chain.net/smartcontract/zcnsc/burn.go

```diff
@@ -39,7 +41,11 @@ func (zcn *ZCNSmartContract) Burn(
 	}
 
 	// check burn amount
-	if state.Balance(trans.Value*1e10) < gn.MinBurnAmount {
+	transSAS, err := currency.ParseZCN(float64(trans.Value))
```

Comment:
Treat txn.Value as SAS, and notify the related callers about this changing.

Reply:
https://github.com/0chain/0chain/issues/1306